network/p2p/connection/shuffler_test.go

```diff
@@ -0,0 +1,113 @@
+package connection_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/network/p2p/connection"
+	p2ptest "github.com/onflow/flow-go/network/p2p/test"
+)
+
+// TestShuffle test that shuffle is not in place and that it returns a different shuffle each time
+// it is called.
+func TestShuffle(t *testing.T) {
+	pid := p2ptest.PeerIdFixture(t)
+
+	s, err := connection.NewPeerIdSliceShuffler(pid)
+	require.NoError(t, err)
+
+	peerIds := p2ptest.PeerIdSliceFixture(t, 10)
+
+	shuffled1 := s.Shuffle(peerIds)
+
+	// shuffle should not be in place, so shuffled1 should not be equal to peerIds
+	require.NotEqual(t, peerIds, shuffled1)
```

Comment:
Based on this test, if the shuffle happens to be identity (with prob 1/10! which is not extremely small), the test would fail.  To check the shuffle isn't done in place (as per the test comment), maybe we could find a way to check the actual pointers?  
`require.NotEqual` won't work for pointer comparison as the doc mentions it always compares the values instead.

Reply:
We could also make `10` higher to make flakiness less frequent (currently, it happens once every 3.5 M tests)

Reply:
With the use of `rand.Shuffle`, the `Shuffler` have been removed.